src/frontend/src/api/backend-sdk/sendToken.ts

```
@@ -0,0 +1,107 @@
+ */
+export async function getCustomerGuid(): Promise<string | null> {
+  try {
+    logMessage('info', 'Fetching Clerk auth token...');
+
+    const authData = auth();
+    const { orgId } = authData;
+    const token = await authData.getToken();
+
+    if (!orgId || !token) {
+      logMessage('error', 'Missing Clerk orgId or token. Authentication failed.');
+      return null;
+    }
+
+    // If orgId exists in cache, return stored GUID (avoid duplicates)
+    if (customerCache.has(orgId)) {
+      logMessage('info', `Customer GUID found in cache for this orgId: ${orgId}`);
+      logCustomerCache(); // Log the cache contents
+      return customerCache.get(orgId)!;
+    }
+
+    // First-time call to API
+    const customerGuid = await fetchCustomerGuid(orgId, token);
+
+    if (customerGuid && !customerCache.has(orgId)) {
+      // Store in cache only if not already stored
+      customerCache.set(orgId, customerGuid);
+      logMessage('info', `Stored new customer GUID in cache for this orgId: ${orgId}`);
+    }
+
+    logCustomerCache(); // Log cache after storing new entry
+
+    return customerGuid;
+  } catch (error) {
+    logMessage('error', 'Error fetching customer GUID', error);
+    return null;
+  }
+}
+
+/**
+ * Fetch customer GUID from backend (called only if not in cache)
+ */
+async function fetchCustomerGuid(orgId: string, token: string): Promise<string | null> {
+  const apiUrl = `http://${CHAT_SERVICE_HOST}:${CHAT_SERVICE_PORT}/addcustomer`;
+
+  try {
+    logMessage('info', `Calling API (only if not in cache): ${apiUrl}`);
+
+    const response = await fetch(apiUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'Authorization': `Bearer ${token}`,
+      },
+      body: JSON.stringify({ orgId }),
+    });
+
+    if (!response.ok) {
+      logMessage('error', `API Error: ${response.status} ${response.statusText}`);
+      return null;
```

You need to throw an exception and display an error page in layout.tsx